pkg/osvscanner/internal/scanners/lockfile.go

```diff
@@ -21,9 +21,8 @@ var lockfileExtractorMapping = map[string]string{
 	"pnpm-lock.yaml":    "javascript/pnpmlock",
 	"yarn.lock":         "javascript/yarnlock",
 	"package-lock.json": "javascript/packagelockjson",
-	// This translation works for both the transitive scanning and non transitive scanning
-	// As both extractors have the same name
-	"pom.xml":                     "java/pomxml",
+	// Use the transitive extractor by default.
+	"pom.xml":                     "java/pomxmlnet",
```

Comment:
I think we need a special case for pom.xml to select the non transitive extractor here if offline scanning is enabled.  (Or even better, we just support offline scanning in pomxmlnet, a hacky way to do this would be to include the pomxml (not net) extractor in the pomxmlnet extractor, and if we don't pass in a dependency client, fallback to that)